### PR TITLE
Mostly refactor

### PR DESCRIPTION
**Pull request by stitchinger**
- scoreboard is decoupled
- removed some unused code
- some council refactor
- added small feature for council

### CODE REVIEW

File: src/main/java/io/georgeous/mcgenerations/council/CouncilListener.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        if (loc.getWorld() != MCG.overworld) return false;
+
+        return !(loc.distance(McgConfig.getCouncilLocation()) > triggerRadius);
+    }
+
+    private boolean hasAllSpawningCriteria(Player player) {
+        ServerPlayer playerWrapper = PlayerManager.get().getWrapper(player);
+        if (playerWrapper == null) return false;
+
+        if (!playerWrapper.getRulesAccepted()) {
```
Comment (Benlewis9000) [inline src/main/java/io/georgeous/mcgenerations/council/CouncilListener.java:95]: Not really part of this PR but bool queries follow a "has" or "is" naming convention e.g. "hasAcceptedRules"

File: src/main/java/io/georgeous/mcgenerations/scoreboard/ScoreboardRefreshListener.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+public class ScoreboardRefreshListener implements Listener {
+
+    ScoreboardHandler handler;
+
+    public ScoreboardRefreshListener(ScoreboardHandler handler){
+        this.handler = handler;
+    }
+
+    @EventHandler (priority = EventPriority.MONITOR)
```
Comment (Benlewis9000) [inline src/main/java/io/georgeous/mcgenerations/scoreboard/ScoreboardRefreshListener.java:25]: Monitor should only ever be used by listeners that don't change anything, e.g. loggers, change this to HIGHEST as it could break other things that only want to listen in

Comment (Benlewis9000) [reply src/main/java/io/georgeous/mcgenerations/scoreboard/ScoreboardRefreshListener.java:25]: Unless there's a specific reason this is monitor

Comment (Benlewis9000) [reply src/main/java/io/georgeous/mcgenerations/scoreboard/ScoreboardRefreshListener.java:25]: and same for all the others, I have a feeling none of them should be Monitor

Comment (Benlewis9000) [reply src/main/java/io/georgeous/mcgenerations/scoreboard/ScoreboardRefreshListener.java:25]: I don't think they even need to be highest, medium or low will be fine, probably lowest since you'd want to trigger a refresh after everything else has made changes as needed

File: src/main/java/io/georgeous/mcgenerations/spawning/SpawnManager.java [hunk 3]
```diff
@@ -78,8 +78,12 @@ private static void startSpawningFeedback(Player spawnedPlayer) {
         spawnedPlayer.getWorld().playSound(spawnedPlayer.getLocation(), Sound.BLOCK_BEACON_POWER_SELECT, SoundCategory.MASTER, 1.5f, 1f);
         spawnedPlayer.getWorld().spawnParticle(Particle.END_ROD, spawnedPlayer.getLocation().add(0, 1, 0), 60, 0.01, 0.05, 0.01, 0.2);
 
-
-        Notification.neutralMsg(spawnedPlayer, "You will be reborn in " + McgConfig.getSecInLobby() + " seconds");
+        spawnedPlayer.sendTitle(
+                "",
+                "You will be reborn in §c" + McgConfig.getSecInLobby() + "§f seconds",
```
Comment (Benlewis9000) [inline src/main/java/io/georgeous/mcgenerations/spawning/SpawnManager.java:83]: Use ChatColor

File: src/main/java/io/georgeous/mcgenerations/systems/family/Family.java [hunk 4]
```diff
@@ -75,7 +77,10 @@ public String getName() {
     public void setName(String name) {
         this.name = name;
 
+        FamilyChangeEvent e = new FamilyChangeEvent(this);
```
Comment (Benlewis9000) [inline src/main/java/io/georgeous/mcgenerations/systems/family/Family.java:80]: If you cancelled this event the name would still be changed

Comment (Benlewis9000) [reply src/main/java/io/georgeous/mcgenerations/systems/family/Family.java:80]: Are you sure you want the event to be cancellable?

Comment (Benlewis9000) [reply src/main/java/io/georgeous/mcgenerations/systems/family/Family.java:80]: Genuine question, I'm not sure

File: src/main/java/io/georgeous/mcgenerations/systems/family/FamilyChangeEvent.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+public class FamilyChangeEvent extends Event implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+
+    private final Family family;
```
Comment (Benlewis9000) [inline src/main/java/io/georgeous/mcgenerations/systems/family/FamilyChangeEvent.java:14]: name suggestion, "newFamily". I'd also suggest passing in the old family and the player who's family has changed, event is a bit useless otherwise, as it the only info it gives you as a family that some person just joined

File: src/main/java/io/georgeous/mcgenerations/scoreboard/ScoreboardRefreshListener.java (hunk 2 above)
Comment (stitchinger) [reply src/main/java/io/georgeous/mcgenerations/scoreboard/ScoreboardRefreshListener.java:25]: Highest Priority gets called the last one. I used monitor, bc the scoreboard kinda only does monitor the values.
It would never want to change values or cancel an event.

File: src/main/java/io/georgeous/mcgenerations/systems/family/Family.java (hunk 4 above)
Comment (stitchinger) [reply src/main/java/io/georgeous/mcgenerations/systems/family/Family.java:80]: No, I dont want it to be cancellable. I copied the code from another event of mine and was too lazy to adjust it xD
